tests/src/test/scala/cats/tests/ListTests.scala

```diff
@@ -42,4 +42,10 @@ class ListTests extends CatsSuite {
       l.show should === (l.toString)
     }
   }
+
+  test("indexed consistent with zipWithIndex") {
+    forAll { (fa: List[String]) =>
+      Traverse[List].indexed(fa) should === (fa.zipWithIndex)
```

Comment:
Added. Went ahead and created a dedicated file for testing `Traverse` instead of piggybacking on the tests in `ListTests.scala`.